src/Uno.UI/UI/Xaml/Controls/ListViewBase/ListViewBaseSource.iOSMacOS.cs

```diff
@@ -0,0 +1,518 @@
+			}
+
+			if (this.Log().IsEnabled(Microsoft.Extensions.Logging.LogLevel.Debug))
+			{
+				this.Log().Debug($"Count requested for section {section}, returning {count}");
+			}
+			return count;
+		}
+
+		private int GetUngroupedItemsCount(nint section)
+		{
+			int count;
+			if (section == 0)
+			{
+				count = Owner.XamlParent.NumberOfItems;
+			}
+			else
+			{
+				// Extra section added to accommodate header+footer, contains no items.
+				count = 0;
+			}
+
+			return count;
+		}
+
+		private int GetGroupedItemsCount(nint section)
+		{
+			if ((int)section >= Owner.XamlParent.NumberOfDisplayGroups)
+			{
+				// Header+footer section which is empty
+				return 0;
+			}
+			return Owner.XamlParent.GetDisplayGroupCount((int)section);
+		}
+#if __IOS__
+		public override UICollectionViewCell GetCell(UICollectionView collectionView, NSIndexPath indexPath)
+#else
+		public override NSCollectionViewItem GetItem(NSCollectionView collectionView, NSIndexPath indexPath)
+#endif
+		{
+			using (
+			   _trace.WriteEventActivity(
+				   TraceProvider.ListViewBaseSource_GetCellStart,
+				   TraceProvider.ListViewBaseSource_GetCellStop
+			   )
+			)
+			{
+				// Marks this item to be materialized, so its actual measured size 
+				// is used during the calculation of the layout.
+				// This is required for paged lists, so that the layout calculation
+				// does not eagerly get all the items of the ItemsSource.
+				UpdateLastMaterializedItem(indexPath);
+
+				var index = Owner?.XamlParent?.GetIndexFromIndexPath(IndexPath.FromNSIndexPath(indexPath)) ?? -1;
+
+				var identifier = GetReusableCellIdentifier(indexPath);
+
+				var listView = (NativeListViewBase)collectionView;
+#if __IOS__
+				var cell = (ListViewBaseInternalContainer)collectionView.DequeueReusableCell(identifier, indexPath);
+#else
+				var cell = (ListViewBaseInternalContainer)collectionView.MakeItem(GetReusableCellIdentifier(indexPath), indexPath);
+#endif
+
+				using (cell.InterceptSetNeedsLayout())
+				{
+					var selectorItem = cell.Content as SelectorItem;
+
+					if (selectorItem == null ||
+						// If it's not a generated container then it must be an item that returned true for IsItemItsOwnContainerOverride (eg an
+						// explicitly-defined ListViewItem), and shouldn't be recycled for a different item.
+						!selectorItem.IsGeneratedContainer)
+					{
+						cell.Owner = Owner;
+						selectorItem = Owner?.XamlParent?.GetContainerForIndex(index) as SelectorItem;
+						cell.Content = selectorItem;
+						if (this.Log().IsEnabled(Microsoft.Extensions.Logging.LogLevel.Debug))
+						{
+							this.Log().Debug($"Creating new view at indexPath={indexPath}.");
+						}
+
+						FrameworkElement.InitializePhaseBinding(selectorItem);
+
+						// Ensure the item has a parent, since it's added to the native collection view
+						// which does not automatically sets the parent DependencyObject.
+						selectorItem.SetParent(Owner?.XamlParent);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['selectorItem' is null on at least one execution path.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=4931087)